data/pigui/views/game.lua

```diff
@@ -277,6 +279,37 @@ Event.Register("onPauseMenuClosed", function()
 	Input.EnableBindings()
 end)
 
+Event.Register("onGamePaused", function()
```

Comment:
These four event handlers should become `ui.Events:Register(..., function() ... end)`. `onGamePaused` et. al. is pushed to the UI event queue and there is no guarantee that events on the main queue will be processed until the pause ends.